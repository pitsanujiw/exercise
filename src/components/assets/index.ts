export * from './assetItem';
export * from './assetsTable';
