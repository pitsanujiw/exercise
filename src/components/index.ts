export * from "./assets";
export * from "./loading";
export * from "./orderForm";
