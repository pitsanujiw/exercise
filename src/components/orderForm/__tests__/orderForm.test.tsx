import { OrderForm } from "components";
import { render } from "@testing-library/react";

describe("OrderForm", () => {
  it("Should render it correctly", async () => {
    const renderResult = render(
      <div>
        <OrderForm submitOrder={() => {}} />
      </div>
    );
    expect(renderResult).toMatchSnapshot();
  });
});
