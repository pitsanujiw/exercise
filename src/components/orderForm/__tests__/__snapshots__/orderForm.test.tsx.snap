// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Should render it correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <form
            class="makeStyles-formControl-1"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input_padding-2"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-side"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="mui-component-select-side"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="side"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                data-shrink="false"
              >
                Price
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input"
                  name="price"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                data-shrink="false"
              >
                Amount
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input"
                  name="amount"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <form
          class="makeStyles-formControl-1"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input_padding-2"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-side"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="mui-component-select-side"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="side"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Price
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="price"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Amount
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="amount"
                type="number"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`OrderForm Should render it correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <form
            class="makeStyles-formControl-1"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input_padding-2"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-side"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="mui-component-select-side"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="side"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                data-shrink="false"
              >
                Price
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input"
                  name="price"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                data-shrink="false"
              >
                Amount
              </label>
              <div
                class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiFilledInput-input"
                  name="amount"
                  type="number"
                  value=""
                />
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <form
          class="makeStyles-formControl-1"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input_padding-2"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-side"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="mui-component-select-side"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              name="side"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Price
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="price"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Amount
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input_padding-2 MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="amount"
                type="number"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
